ingest-app/src/cmr/ingest/data/bulk_update.clj

```diff
@@ -41,70 +42,71 @@
+              (update :request-json-body util/gzip-blob->string)
+              (update :created-at (partial oracle/oracle-timestamp->str-time conn)))))
 
   (get-bulk-update-task-collection-status
     [db task-id]
     ;; Get statuses for all collections by task id
     (map util/map-keys->kebab-case
-      (su/query db (su/build (su/select [:concept-id :status :status-message]
-                              (su/from "bulk_update_coll_status")
-                              (su/where `(= :task-id ~task-id)))))))
+         (su/query db (su/build (su/select [:concept-id :status :status-message]
+                                           (su/from "bulk_update_coll_status")
+                                           (su/where `(= :task-id ~task-id)))))))
 
   (get-bulk-update-collection-status
     [db task-id concept-id]
     ;; Get the status for a particular collection
     (su/find-one db (su/select [:status :status-message]
-                     (su/from "bulk_update_coll_status")
-                     (su/where `(and (= :task-id ~task-id)
-                                     (= :concept-id ~concept-id))))))
+                               (su/from "bulk_update_coll_status")
+                               (su/where `(and (= :task-id ~task-id)
+                                               (= :concept-id ~concept-id))))))
 
   (create-and-save-bulk-update-status
     [db provider-id json-body concept-ids]
     ;; In a transaction, add one row to the task status table and for each concept
     (try
-     (j/with-db-transaction
-      [conn db]
-      (let [task-id (:nextval (first (su/query db ["SELECT task_id_seq.NEXTVAL FROM DUAL"])))
-            statement (str "INSERT INTO bulk_update_task_status "
-                           "(task_id, provider_id, request_json_body, status)"
-                           "VALUES (?, ?, ?, ?)")
-            values [task-id provider-id (util/string->gzip-bytes json-body) "IN_PROGRESS"]]
-       (j/db-do-prepared db statement values)
-       ;; Write a row to collection status for each concept id
-       (apply j/insert! conn
-              "bulk_update_coll_status"
-              ["task_id" "concept_id" "status"]
-              ;; set :transaction? false since we are already inside a transaction
-              (concat (map #(vector task-id % "PENDING") concept-ids) [:transaction? false]))
-       task-id))
-     (catch Exception e
-      (errors/throw-service-error :invalid-data
-       [(str "Error creating creating bulk update status "
-             (.getMessage e))]))))
+      (j/with-db-transaction
+        [conn db]
+        (let [task-id (:nextval (first (su/query db ["SELECT task_id_seq.NEXTVAL FROM DUAL"])))
+              statement (str "INSERT INTO bulk_update_task_status "
+                             "(task_id, provider_id, name, request_json_body, status)"
+                             "VALUES (?, ?, ?, ?, ?)")
+              name (or (get (json/parse-string json-body) "name") task-id) 
```

Comment:
`get` accepts an additional parameter of a default value when not found in the map. We should use that rather than or.

```
(get (json/parse-string json-body) "name" task-id)
```